client/signup/controller.js

```diff
@@ -284,25 +291,26 @@ export default {
 			flow: flowName,
 		} );
 
-		context.store.dispatch( setLayoutFocus( 'content' ) );
-		context.store.dispatch( setCurrentFlowName( flowName ) );
+		dispatch( setLayoutFocus( 'content' ) );
+		dispatch( setCurrentFlowName( flowName ) );
 
 		if ( ! [ 'launch-site' ].includes( flowName ) ) {
-			context.store.dispatch( setSelectedSiteId( null ) );
+			dispatch( setSelectedSiteId( null ) );
 		}
 
 		context.primary = createElement( SignupComponent, {
-			store: context.store,
-			path: context.path,
+			store: signupStore,
+			path: path,
 			initialContext,
-			locale: context.params.lang,
+			locale: params.lang,
 			flowName,
 			queryObject: query,
 			refParameter: query && query.ref,
 			stepName,
 			stepSectionName,
 			stepComponent,
 			pageTitle: getFlowPageTitle( flowName, userLoggedIn ),
+			siteIdOrSlug: getSideIdOrSlug( signupDependencies, query ),
```

Comment:
hmm It looks to me like passing `siteIdOrSlug` here make[ the code in `setSelectedSiteForSignup`](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/controller.js#L310) unnecessary? I wonder if the code in there is still used by other flows or can we delete that whole function.

Also the `setSelectedSiteForSignup` function handles [a case where the site is not yet available](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/controller.js#L327) and fetches the site with `dispatch( requestSite( siteIdOrSlug ) )`. This implementation does not handle that case. Although I couldn't reproduce the condition where that code gets excecuted 🤔

Reply:
hmm, I think `setSelectedSiteForSignup` still has its usage even though the result would be clear. For example, if we cannot get the correct site id from `getSiteId` selector, it would request API to fetch the site info, then we can get the correct site id from `getSiteId` next time. But if we remove that function, we cannot get the correct result. Thus, I prefer to keep that function and avoid clearing site id if the flow with `siteId` or `siteSlug` as dependency.

Reply:
hmm I had another thought, do you think that passing in `siteIdOrSlug` here is still neccessary now that we have made the change to not clear `slectedSiteId` if it is included in `providesDependenciesInQuery`